### ISSUE TITLE
Instrumenting function calls with symbol names

### ISSUE THREAD

**Issue by qinkunbao** [S2E/s2e-env#433]
Hi,

I am reading the document from

http://s2e.systems/docs/Howtos/LuaInstrumentation.html

It demonstrates an example on instrumenting a function by its address. I was wondering if the same goal can
be achieved with a symbol name. It will be a very useful feature for us to analyze some binaries with debugging information included. Some other binary instrument tools (e.g., Intel Pin) also offer such functionalities.

**Comment by vitalych** [comment]
Yes, it would be a good addition. You could add a "function_name" to the configuration and resolve it to a program counter here [1]. To resolve this, you will need to load the binary file and extract the exported functions using the Vmi plugin [2]. The `vmi::ExecutableFile` class lets you extract exported functions (works only for Windows PE files currently [3]).

[1] https://github.com/S2E/s2e/blob/master/libs2eplugins/src/s2e/Plugins/Lua/LuaFunctionInstrumentation.cpp#L125
[2] https://github.com/S2E/s2e/blob/master/libs2eplugins/src/s2e/Plugins/Core/Vmi.h#L70
[3] https://github.com/S2E/s2e/blob/master/libvmi/include/vmi/PEFile.h#L126

**Comment by qinkunbao** [comment]
Thx
